### PR TITLE
Save Plans Modal

### PR DESCRIPTION
**Pull request by imanmoreira**
Modal to allow users to create new plans off of several different options:

Capabilities:
- Create new blank plan
- Create new plan based on example schedule if they have a coop-cycle selected
- Create new plan based on existing plan
- Create new plan based on an advisor excel plan
- Error handling for if user does not fill out a plan name or select a plan type
- Error handling for if a user chooses a duplicate name

### CODE REVIEW

File: frontend/src/Onboarding/OnboardingInfoScreen.tsx [hunk 1]
```diff
@@ -143,7 +138,7 @@ class OnboardingScreenComponent extends React.Component<
 
   onChangeGradYear(e: any) {
     this.setState({
-      gradYear: Number(e.target.value),
+      gradYear: e.target.value == 2020 ? Number(e.target.value) : 2020,
```
Comment (NilsBacke) [inline frontend/src/Onboarding/OnboardingInfoScreen.tsx:141]: What's the reason for this?

Comment (imanmoreira) [reply frontend/src/Onboarding/OnboardingInfoScreen.tsx:141]: Oh shoot I was trying to test something and forgot to remove that

**Comment by ArunJ98** [comment]
image
image
Some small UI issues, the border of the input box overlaps the text and the dropdown covers the input box. These are probably some whack material UI issues :(

**Review by ArunJ98 — COMMENTED** [review]
Code looks great! comment above for some UI issues

**Comment by NilsBacke** [comment]
Looks like there are some conflicts, otherwise all looks good

**Review by NilsBacke — APPROVED** [review]

**Comment by NilsBacke** [comment]
[Create Plan Modal](https://trello.com/c/krjCSOSu/148-create-plan-modal)

**Pull request merged**